File: codeforces/practices/B_QueryAboutLessOrEqualsElements/TaskB.cpp
#include <iostream>
#include <cstdio>
#include <cstdlib>
#include <algorithm>

using namespace std;

int main() {
  int n;
  int m;
  int a[200000];
  int b[200000];
  scanf("%d%d", &n, &m);
  for (int i = 0; i < n; i++) {
    scanf("%d", &a[i]);
  }
  for (int i = 0; i < m; i++) {
    scanf("%d", &b[i]); 
  }
  sort(a + 0, a + n);
  for (int i = 0; i < m; i++) {
    if (b[i] >= a[n - 1]) {
      printf("%d ", n);
      continue;
    }
    int l = 0; 
    int r = n - 1;
    int result = 0;
    while (l <= r) {
      int mid = (l + r) / 2;
      if (a[mid] > b[i]) {
        result = mid;
        r = mid - 1;
      } else {
        l = mid + 1;
      }
    }
    printf("%d ", result);
  }
  printf("\n"); 
  return 0;
}

File: codeforces/practices/B_QueryAboutLessOrEqualsElements/TaskB.java
import java.io.PrintWriter;
import java.util.Arrays;
import java.util.Scanner;

public class TaskB {

  public static void main(String[] args) {
    TaskB tb = new TaskB();
    PrintWriter pw = new PrintWriter(System.out);
    tb.solve(new Scanner(System.in), pw);
    pw.close();
  }

  public void solve(Scanner input, PrintWriter output) {
    int n = input.nextInt();
    int m = input.nextInt();
    
    int[] a = new int[n];
    for (int i = 0; i < n; i++) {
      a[i] = input.nextInt();
    }
    int[] b = new int[m];
    for (int i = 0; i < m; i++) {
      b[i] = input.nextInt();
    }

    Arrays.sort(a);
    for (int num : b) {
      int idx = getFirstLargerIdx(a,num);
      output.print(idx + " ");
    }
    output.println();
  }

  private int getFirstLargerIdx(int[] arr, int target) {
    if (arr[arr.length - 1] <= target) {
      return arr.length;
    }
    int l = 0;
    int r = arr.length - 1;
    int result = 0;
    while (l <= r) {
      int mid = (l + r) / 2;
      if (arr[mid] > target) {
        result = mid;
        r = mid - 1;
      } else {
        l = mid + 1;
      }
    }
    return result;
  }
}
